File: frontend/src/dtos/tier.ts
import { z } from "zod";
import { nameSchema, nullableUrlSchema } from "@utils/dto";

export const AddTierSchema = z.object({
  name: nameSchema,
  iconUrl: nullableUrlSchema,
});

export const UpdateTierSchema = AddTierSchema.partial();

export type AddTierDTO = z.infer<typeof AddTierSchema>;
export type UpdateTierDTO = z.infer<typeof UpdateTierSchema>;

export interface TierDTO {
  tierId: number;
  presetId: number;
  name: string;
  iconUrl: string | null;
}

File: python/shared/dtos/tier.py
from . import BaseDTO, BaseEntityDTO, NameStr, NullableUrlStr


class TierDTO(BaseEntityDTO):
    tier_id: int
    preset_id: int
    name: str
    icon_url: str | None


class AddTierDTO(BaseDTO):
    name: NameStr
    icon_url: NullableUrlStr


class UpdateTierDTO(BaseDTO):
    name: NameStr | None = None
    icon_url: NullableUrlStr = None
